common/djangoapps/student/helpers.py

```diff
@@ -278,3 +290,114 @@ def destroy_oauth_tokens(user):
     dop_refresh_token.objects.filter(user=user.id).delete()
     dot_access_token.objects.filter(user=user.id).delete()
     dot_refresh_token.objects.filter(user=user.id).delete()
+
+
+class AccountValidationError(Exception):
+    """
+    Used in account creation views to raise exceptions with details about specific invalid fields
+    """
+    def __init__(self, message, field):
+        super(AccountValidationError, self).__init__(message)
+        self.field = field
+
+
+def create_or_set_user_attribute_created_on_site(user, site):
+    """
+    Create or Set UserAttribute indicating the microsite site the user account was created on.
+    User maybe created on 'courses.edx.org', or a white-label site. Due to the very high
+    traffic on this table we now ignore the default site (eg. 'courses.edx.org') and
+    code which comsumes this attribute should assume a 'created_on_site' which doesn't exist
+    belongs to the default site.
+    """
+    if site and site.id != settings.SITE_ID:
+        UserAttribute.set_user_attribute(user, 'created_on_site', site.domain)
+
+
+
+def do_create_account(form, custom_form=None):
```

Comment:
does ficus not have this function already? maybe it was called `_do_create_account` back then